### PR TITLE
Add permissions by bit array

### PR DESCRIPTION
**Pull request by bennyman123abc**

### CODE REVIEW

File: src/Structs/Permissions.js [hunk 1]
```diff
@@ -9,16 +9,31 @@ class Permissions {
...
     permissions = 0
 
-    constructor (bit) {
-        this.permissions = bit
+    constructor (bitfield) {
+        if (Array.isArray(bitfield))
+            this.set(bitfield)
+        else if (typeof bitfield === "number")
+            this.permissions = bitfield
+    }
```
Comment (matthe815) [inline src/Structs/Permissions.js:17]: Make the isArray a single line "if this then nothing else"-type of method with the else-if being on it's own at the bottom.

File: src/Structs/Permissions.js [hunk 2]
```diff
@@ -9,16 +9,31 @@ class Permissions {
...
+
+    /**
+     * Set the permissions bitfield.
+     * @param {Array<Permission>} permissions 
+     */
+    set (permissions) {
+        this.permissions = 0
+        permissions.forEach(permission => this.add(permission))
     }
```
Comment (matthe815) [inline src/Structs/Permissions.js:26]: Remove `this.permissions = 0` as bitwise does not reapply permissions.

File: src/Structs/Permissions.js [hunk 3]
```diff
@@ -9,16 +9,31 @@ class Permissions {
...
      * Add a new permission bit to the user.
-     * @param {Permission} permission
+     * @param {Permission|Array<Permission>} permission
      */
     add (permission) {
-        this.permissions == this.permissions | permission
+        if (Array.isArray(permission))
+            permission.forEach(p => this.permissions = this.permissions | p);
+        else 
+            this.permissions == this.permissions | permission;
```
Comment (matthe815) [inline src/Structs/Permissions.js:36]: Similar to the first comment, single-returned-line for the isArray.

**Review by matthe815 — CHANGES_REQUESTED** [review]

**Review by bennyman123abc — COMMENTED** [review]
Yes

**Review by matthe815 — APPROVED** [review]

**Pull request merged**